python/tank/bootstrap/configuration.py

```diff
@@ -92,10 +92,20 @@ def get_tk_instance(self, sg_user):
         # by creating a pipeline configuration object directly
         # and pass that into the factory method.
 
-        pc = pipelineconfig.PipelineConfiguration(path, self.descriptor)
+        if self._does_pipelineconfigution_supports_descriptor(pipelineconfig.PipelineConfiguration):
+            pc = pipelineconfig.PipelineConfiguration(path, self.descriptor)
```

Comment:
Yeah, I'm just trying to think of ways to not change the API. It feels like in this case, since `descriptor` is optional and there are clearly legacy cases where it's not required, perhaps a setter isn't too horrible? Maybe there's some other way to expose the descriptor outside of the pipeline config? I don't really have a good overall grasp of all the pieces, so feel free to ignore me. :)